extensions/lifecycle/lifecycleProducer/LifecycleProducer.js

```diff
@@ -0,0 +1,256 @@
+'use strict'; // eslint-disable-line
+
+const async = require('async');
+const AWS = require('aws-sdk');
+const http = require('http');
+const Logger = require('werelogs').Logger;
+
+const BackbeatProducer = require('../../../lib/BackbeatProducer');
+const BackbeatConsumer = require('../../../lib/BackbeatConsumer');
+const AccountCredentials =
+    require('../../../lib/credentials/AccountCredentials');
+const safeJsonParse = require('../util/safeJsonParse');
+
+const CONSUMER_FETCH_MAX_BYTES = 5000020;
+
+class LifecycleProducer {
+    /**
+     * Constructor of LifecycleProducer
+     * @param {Object} zkConfig - zookeeper config
+     * @param {Object} lcConfig - lifecycle config
+     * @param {Object} s3Config - s3 config
+     * @param {String} s3Config.host - host ip
+     * @param {String} s3Config.port - port
+     * @param {Object} authConfig - auth config
+     * @param {String} authConfig.account - account name
+     * @param {Object} authConfig.vault - vault details
+     * @param {String} authConfig.vault.host - vault host ip
+     * @param {number} authConfig.vault.port - vault port
+     * @param {number} authConfig.vault.adminPort - vault admin port
+     * @param {String} transport - http or https
+     */
+    constructor(zkConfig, lcConfig, s3Config, authConfig, transport) {
+        this._log = new Logger('Backbeat:LifecycleProducer');
+        this._zkConfig = zkConfig;
+        this._lcConfig = lcConfig;
+        this._s3Client = new AWS.S3({
+            endpoint: `${transport}://${s3Config.host}:${s3Config.port}`,
+            credentials: new AccountCredentials(authConfig, this._log),
+            sslEnabled: transport === 'https',
+            s3ForcePathStyle: true,
+            signatureVersion: 'v4',
+            httpOptions: {
+                agent: new http.Agent({ keepAlive: true }),
+                timeout: 0,
+            },
+            maxRetries: 0,
+        });
+        this._bucketProducer = null;
+        this._objectProducer = null;
+        this._bucketConsumer = null;
+        this._internalTaskScheduler = async.queue((ctx, cb) => {
+            const { task, rules, value } = ctx;
+            return task.processBucketEntry(rules, value, cb);
+        }, this._lcConfig.producer.concurrency);
+
+        this._internalTaskScheduler.drain(err => {
+            if (err) {
+                this._log.error('error occured when queue is drained', {
+                    error: err,
+                });
+            }
+        });
+    }
+
+    /**
+     * Send entry to the object task topic
+     * @param {[type]} entry - The Kafka entry to send to the topic
+     * @param {Function} cb - The callback to call
+     * @return {undefined}
+     */
+    sendObjectEntry(entry, cb) {
+        const strEntry = JSON.stringify(entry);
+        this._objectProducer.send([strEntry], cb);
```

Comment:
I think this will not work, because the `send` method expects an array of *objects*, where each object has a `message` attribute and optionally a `key` attribute as well. So normally the proper way would be to construct an array of objects with a `message` attribute which is the result of `JSON.stringify`, if I'm not mistaken.

Reply:
So just making sure, but would it then be something like `[ { message: strEntry } ]`

Reply:
yes

Reply:
Updated.